plugins/modules/nmcli.py

```diff
@@ -1832,7 +1832,7 @@ def connection_options(self, detect_change=False):
         elif self.type == 'wifi':
             options.update({
                 '802-11-wireless.ssid': self.ssid,
-                'connection.slave-type': 'bond' if self.master else None,
+                'connection.slave-type': self.slave_type if self.master else None,
```

Comment:
By default `self.slave_type` is `None`. Shouldn't this be changed so that `bond` is used if `slave_type` hasn't been set?

Reply:
Assume that you want to use the module to create a wifi hostspot bridged, you can configure the task like I did:
```yaml
- name: Configure hotspot
  community.general.nmcli:
    type: wifi
    conn_name: foo
    ifname: wlan0
    slave_type: bridge
    master: br0
    ssid: bar
    wifi:
      mode: ap
    wifi_sec:
      key-mgmt: wpa-psk
      psk: foobar
      proto:
        - rsn
      pairwise:
        - ccmp
    state: "{{ 'present' if item.enabled else 'absent' }}"
```
but, if we leave `bond` wired in the code, the connection doesn't work, like the examples described in the issue, because the master interface is a `bridge` in this case.
So I thinks it's better to leave the user type the `slave-type` according to the interface set in the `master`, to bond, bridge or team.
Maybe a new feature may be checking the type of the `master` interface and set accordingly and automatically, but until that, you need to permit the user set the `slave_type` parameter.

Thanks

Reply:
You could also simply modify the code to fall back to `bond` if `self.slave_type is None`. Otherwise this is a breaking change and won't get merged.

Reply:
Hi, you mean like that?
```suggestion
                'connection.slave-type': 'bond' if self.slave_type is None else self.slave_type,
```
I assume that parameter is wired to 'bond' because is mandatory in case of a wifi connection without a master interface. I need to search a doc for this case, if it's not necessary the line could be removed, the association is already present on line 1726

```python
                'connection.slave-type': self.slave_type,
```

Thanks

Reply:
Shouldn't it be rather
```suggestion
                'connection.slave-type': ('bond' if self.slave_type is None else self.slave_type) if self.master else None,
```
?

Alternatively
```suggestion
                'connection.slave-type': (self.slave_type or 'bond') if self.master else None,
```
will also work (assuming that an empty string is not a valid value that needs to be passed on as-is).

Reply:
Hi and Happy New Year.
I prefer your first suggestion but I need to use some of my spare time to check the documentation, maybe all of this logic is not necessary.
Anyway this is a solution to the actual bug.

Thanks

Reply:
Hi, unfortunately I can't find a specific doc that describes this situation, I don't have at this time enough effort to spend. But the solution is working and fixes the bug, if you want to merge the PR
Thanks

Reply:
Please fix the PR first so it won't break existing setups that specify `master` but not `slave_type`.